compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java

```diff
@@ -54,10 +55,18 @@ public class FileSystemCache extends CompilationCache {
     private final Path cacheDirPath;
     private Path birPath;
     private Path packageCacheDirPath;
+    private static Path buildProjectCacheDirPath;
 
     public FileSystemCache(Project project, Path cacheDirPath) {
         super(project);
         this.cacheDirPath = cacheDirPath;
+        if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
+            updateBuildProjectCachePath(cacheDirPath);
+        }
+    }
+
+    public static void updateBuildProjectCachePath(Path cacheDirPath) {
```

Comment:
This method seems to have only one usage which is within the same class (L64). We can make this private in that case.

Reply:
Fixed in b77a6251e986d5d8fd08f85f67d50c0153c8bc32.